controllers/routing/routing_delete_resources.go

```diff
@@ -33,14 +34,14 @@ func (r *PlatformRoutingController) HandleResourceDeletion(ctx context.Context,
 }
 
 func (r *PlatformRoutingController) deleteOwnedResources(ctx context.Context, target *unstructured.Unstructured, gvkList []schema.GroupVersionKind) error {
-	ownerName := target.GetName()
-	ownerKind := target.GetObjectKind().GroupVersionKind().Kind
-	ownerUID := string(target.GetUID())
+	ownerName := labels.OwnerName(target.GetName())
+	ownerKind := labels.OwnerKind(target.GetObjectKind().GroupVersionKind().Kind)
+	ownerUID := labels.OwnerUID(target.GetUID())
 
 	resourceOwnerLabels := client.MatchingLabels{
```

Comment:
Could we get a helper alla: labels.Match(labels...) client.MatchingLabels here somewhere?

Constructing the label type and "manually" constructing the matcher based on the key and value that "hopefully" match (easy copy paste error) isn't too far of the way it already worked.

Reply:
Yeah I was thinking about it too today when looking at your latest PR. Will do

Reply:
So now it's

```golang
labels.MatchingLabels(
			labels.OwnerName(target.GetName()),
			labels.OwnerKind(target.GetObjectKind().GroupVersionKind().Kind),
			labels.OwnerUID(target.GetUID()),
		)
````

Reply:
Thanks for the idea